cmake/modules/OCApplyCommonSettings.cmake

```diff
@@ -15,6 +15,12 @@ function(apply_common_target_settings_soft targetName)
         # this includes deprecations
         # https://learn.microsoft.com/en-us/cpp/build/reference/compiler-option-warning-level?view=msvc-170
         target_compile_options(${targetName} PRIVATE /W3)
+        target_compile_definitions(${targetName}
+            # Get APIs from from Win8 onwards.
+            PUBLIC _WIN32_WINNT=_WIN32_WINNT_WIN8
```

Comment:
Do they need to be public now that we apply them to all our targets?
I think we should also move NOMINMAX .
I'm not sure what the mingw define does and we no longer support mingw....